### PR TITLE
Port ListBuckets & cleanup

### PR DESCRIPTION
**Pull request by klauspost**
## Description

Port the final peerS3 endpoint.

## Motivation and Context

Finishing up some of the last relevant endpoints.

## How to test this PR?

Mint.

## Types of changes
- [x] Optimization (provides speedup with no functional changes)

### CODE REVIEW

**Comment by harshavardhana** [comment]
BTW the mint failure looks like is a PR issue here, it is not reproduced in any other PR.

**Comment by harshavardhana** [comment]
> BTW the mint failure looks like is a PR issue here, it is not reproduced in any other PR.

The type of the setup is compressed + encrypted that is failing consistently with minio-js functional tests with the latest release.

File: cmd/peer-s3-server.go [hunk 1]
```diff
@@ -38,12 +36,7 @@ const (
 )
 
 const (
-	peerS3MethodHealth        = "/health"
-	peerS3MethodMakeBucket    = "/make-bucket"
-	peerS3MethodGetBucketInfo = "/get-bucket-info"
-	peerS3MethodDeleteBucket  = "/delete-bucket"
-	peerS3MethodListBuckets   = "/list-buckets"
-	peerS3MethodHealBucket    = "/heal-bucket"
+	peerS3MethodHealth = "/health"
```
Comment (harshavardhana) [inline cmd/peer-s3-server.go]: Would it even make sense to have this? peer-s3 is no longer needed.

**Review by harshavardhana — COMMENTED** [review]
sent the relevant changes to remove peerS3Server completely.

**Comment by klauspost** [comment]
Found the issue.

**Review by harshavardhana — APPROVED** [review]

**Review by poornas — APPROVED** [review]

**Pull request merged**